src/Firebase/Auth.php

```diff
@@ -479,4 +479,14 @@ public function unlinkProvider($uid, $provider): UserRecord
 
         return $this->getUser($uid);
     }
+
+    public function linkWithOauthCredentials($provider, string $token, bool $isAccessToken = true)
```

Comment:
Splitting this into two methods named

- `linkProviderThroughAccessToken($provider, string $accessToken)`
- `linkProviderThroughIdToken($provider, string $idToken)`

currently seems like a good idea to me - on the one hand, we can then remove the third bool parameter, on the other hand this makes the method names a little more "speaking" and aligns well with the `unlinkProvider()` method.